File: ide-features-trainer/src/training/actions/NextLessonAction.java
package training.actions;

import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.wm.ToolWindowManager;
import training.ui.LearnToolWindowFactory;
import training.ui.views.LearnPanel;

import java.awt.*;

/**
 * Created by jetbrains on 29/07/16.
 *
 *  Skip lesson or go to the next lesson
 *
 *  Shortcuts:
 *      win: alt + shift + right
 *      mac: ctrl + shift + right
 *      (see the plugin.xml to change shortcuts)
 */
public class NextLessonAction extends AnAction{

    @Override
    public void actionPerformed(AnActionEvent e) {
        //check if the lesson view is activate
        final ToolWindowManager toolWindowManager = ToolWindowManager.getInstance(e.getProject());
        Component view = LearnToolWindowFactory.getMyLearnToolWindow().getScrollPane().getViewport().getView();

        if (view instanceof LearnPanel) {
            //click button to skip or go to the next lesson
            LearnPanel learnPanel = (LearnPanel) view;
            learnPanel.clickButton();
        }
    }
}

File: ide-features-trainer/src/training/actions/NextLessonAction.kt
// Copyright 2000-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
package training.actions

import com.intellij.icons.AllIcons
import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import training.learn.CourseManager
import training.ui.LearnToolWindowFactory

class NextLessonAction : AnAction(AllIcons.Actions.Forward) {
  override fun actionPerformed(e: AnActionEvent) {
    val project = e.project ?: return
    val lesson = LearnToolWindowFactory.learnWindowPerProject[project]?.learnPanel?.lesson ?: return
    val lessonsForModules = CourseManager.instance.lessonsForModules
    val index = lessonsForModules.indexOf(lesson)
    if (index < 0 || index >= lessonsForModules.size) return
    CourseManager.instance.openLesson(project, lessonsForModules[index + 1])
  }

  override fun update(e: AnActionEvent) {
    val project = e.project
    val lesson = LearnToolWindowFactory.learnWindowPerProject[project]?.learnPanel?.lesson
    e.presentation.isEnabled = lesson != null && CourseManager.instance.lessonsForModules.lastOrNull() != lesson
  }
}
